libs/core/src/lib/wizard/wizard.component.ts

```diff
@@ -0,0 +1,211 @@
+    @HostListener('window:resize')
+    resizeHandler(): void {
+        const wizardWidth = this._elRef.nativeElement.getBoundingClientRect().width;
+        if (!this._previousWidth || wizardWidth <= this._previousWidth) {
+            this._wizardShrinking();
+        } else if (wizardWidth > this._previousWidth) {
+            this._wizardGrowing();
+        }
+        this._previousWidth = wizardWidth;
+    }
+
+    /** @hidden */
+    ngAfterViewInit(): void {
+        setTimeout(() => {
+            // fixes ExpressionChangedAfterItHasBeenCheckedError
+            this._setContentTemplate();
+            this._subscriptions.add(
+                this.steps.changes.subscribe(() => {
+                    this._handleStepOrStatusChanges();
+                })
+            );
+            this.steps.forEach((step) => {
+                this._setupStepEvents(step);
+            });
+            this._cdRef.detectChanges();
+        });
+    }
+
+    /** @hidden */
+    ngOnDestroy(): void {
+        this._subscriptions.unsubscribe();
+    }
+
+    /** @hidden */
+    private _setupStepEvents(step: WizardStepComponent): void {
+        this._subscriptions.add(
+            step.stepClicked.subscribe((event) => {
+                this._stepClicked(event);
+            })
+        );
+        this._subscriptions.add(
+            step.statusChange.subscribe(() => {
+                this._handleStepOrStatusChanges();
+            })
+        );
+        // need to call wizardShrinking for each step < 168px on first load
+        if (step.wizardLabel && step.elRef.nativeElement.clientWidth < STEP_MIN_WIDTH) {
+            this._wizardShrinking();
+        }
+    }
+
+    /** @hidden */
+    private _wizardShrinking(): void {
+        this.steps.forEach((step) => {
+            if (step.status === 'active' || step.status === 'current') {
+                const currentStep = step;
+                if (step.wizardLabel && step.elRef.nativeElement.clientWidth < STEP_MIN_WIDTH) {
+                    this._hideSomeStep(currentStep);
+                }
+            }
+        });
+    }
+
+    /** @hidden */
+    private _wizardGrowing(): void {
+        this._shrinkWhileAnyStepIsTooNarrow();
+    }
+
+    /** @hidden */
+    private _setContentTemplate(): void {
+        this.steps.forEach((step) => {
+            step.finalStep = false;
+            if (step.status === 'current' && step.content) {
+                this.contentTemplate = step.content.contentTemplate;
+            }
+        });
+        this.steps.last.finalStep = true;
+    }
+
+    /** @hidden */
+    private _hideSomeStep(currentStep: WizardStepComponent): void {
+        // If a small step was found, get the step with a visible label furthest away from the current step and hide the label
+        let stepsArray = this.steps.toArray();
+        stepsArray = stepsArray.filter((step) => {
+            return !step.elRef.nativeElement.classList.contains(STEP_NO_LABEL_CLASS);
```

Comment:
Let's hide `step.elRef.nativeElement` and use some convinient public API for that.
For example `step.hasLabel()` or somethnig similar